auto_submit/lib/server/api_request_handler.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:typed_data';
+
+import 'package:meta/meta.dart';
+import 'package:shelf/shelf.dart';
+
+import '../model/google/token_info.dart';
+import '../request_handling/authentication.dart';
+import '../requests/exceptions.dart';
+import 'request_handler.dart';
+import '../service/config.dart';
+
+/// A [RequestHandler] that handles API requests.
+///
+/// API requests adhere to a specific contract, as follows:
+///
+///  * All requests must be authenticated per [AuthenticationProvider].
+///
+/// `T` is the type of object that is returned as the body of the HTTP response
+/// (before serialization). Subclasses whose HTTP responses don't include a
+/// body should extend `RequestHandler<Body>` and return null in their service
+/// handlers ([get] and [post]).
+@immutable
+abstract class ApiRequestHandler extends RequestHandler {
+  /// Creates a new [ApiRequestHandler].
+  const ApiRequestHandler({
+    required Config config,
+    required this.authenticationProvider,
+    this.requestBodyValue,
+  }) : super(config: config);
+
+  /// Service responsible for authenticating this [Request].
+  final AuthenticationProvider authenticationProvider;
+
+  /// Throws a [BadRequestException] if any of [requiredParameters] is missing
+  /// from [requestData].
+  @protected
+  void checkRequiredParameters(List<String> requiredParameters) {
```

Comment:
nit: rm what isn't used

Reply:
Deleted unused logic in api_request_handler.